src/vs/editor/contrib/stickyScroll/browser/stickyScrollController.ts

```diff
@@ -195,6 +197,26 @@ export class StickyScrollController extends Disposable implements IEditorContrib
 		this._editor.focus();
 	}
 
+	private _updateSpecialLineHeightListener(): void {
+		const model = this._editor.getModel();
+		if (!model) {
+			this._specialLineHeightListener.clear();
+			return;
+		}
+		this._specialLineHeightListener.value = model.onDidChangeSpecialLineHeight((e) => {
+			e.changes.forEach((change) => {
+				if (change.ownerId !== this._editor.getIdNumber() && change.ownerId !== 0) {
+					return;
+				}
+				const lineNumber = change.lineNumber;
+				const indexOfLineNumber = this._widgetState.startLineNumbers.indexOf(lineNumber);
+				if (indexOfLineNumber !== -1) {
+					this._renderStickyScroll(lineNumber);
+				}
+			});
+		});
```

Comment:
Instead of exposing `getIdNumber()` on the editor, the better way of encapsulating this problem is that the editor re-emits the model event and the editor itself filters out all changes to only contain changes which affect itself. Then, clients such as sticky scroll, can listen to the editor event and not need to deal with `ownerId`.